code/modules/mob/transform_procs.dm

```diff
@@ -33,14 +33,59 @@
 	for(var/obj/item/W in src)
 		drop_from_inventory(W)
 	set_species(species.primitive_form)
-	dna.SetSEState(MONKEYBLOCK,1)
-	dna.SetSEValueRange(MONKEYBLOCK,0xDAC, 0xFFF)
+	if(!kpg)
+		dna.SetSEState(MONKEYBLOCK,1)
+		dna.SetSEValueRange(MONKEYBLOCK,0xDAC, 0xFFF)
 
 	src << "<B>You are now [species.name]. </B>"
 	qdel(animation)
 
 	return src
 
+/mob/living/carbon/human/proc/humanize(kpg) // we needed this a lot to be honest, why wasn't it made before?
+	if (transforming)
+		return
+	for(var/obj/item/W in src)
+		drop_from_inventory(W)
+	regenerate_icons()
+	transforming = 1
+	canmove = 0
+	stunned = 1
+	icon = null
+	invisibility = 101
+	for(var/t in organs)
+		qdel(t)

```

Comment:
You probably want to cut the organ list too to remove the dangling references that will prevent GC.